packages/backend/src/execution/executors/QueryProviderExecutor.ts

```diff
@@ -1,22 +1,73 @@
 import { QueryProvider } from '../../QueryProvider';
 import { AnyQuery } from '../../types';
-import { QueryNode } from '../../query/createQueryTree';
 import { RefContext, createRefContext } from '../../refs/RefContext';
 import { QueryExecutor } from '../types';
 import { ColumnRef } from '../../refs/ColumnRef';
+import { Table, Where } from '@synthql/queries';
 
-export class QueryProviderExecutor implements QueryExecutor {
-    private providersByTable: Map<string, QueryProvider>;
-    constructor(providers: QueryProvider[]) {
+export class QueryProviderExecutor<DB> implements QueryExecutor {
+    private providersByTable: Map<string, QueryProvider<DB, Table<DB>>>;
+
+    constructor(providers: QueryProvider<DB, Table<DB>>[]) {
         this.providersByTable = new Map(providers.map((p) => [p.table, p]));
     }
 
+    extract(where: Where<DB, Table<DB>>) {
+        const w: Record<string, any> = {};
```

Comment:
I see you're going into a minimalist coding style now 😅 

Would be nice if you could expand `w` into something more readable 😉